eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKitSpec.scala

```diff
@@ -0,0 +1,320 @@
+import org.scalatest.exceptions.TestFailedException
+
+import scala.util._
+
+case class Ping(i: Int)
+
+case class Pong(i: Int)
+
+case class Request(s: String)
+
+case class RequestWithPersistence(s: String, persistenceFirst: Boolean)
+
+case class Response(s: String)
+
+case class PersistSlowly(remaining: Seq[Int])
+
+class SimpleActor(log: ActorRef) extends EventsourcedActor with PersistOnEvent {
+
+  override def id: String = "simple-actor"
+
+  override def eventLog: ActorRef = log
+
+  var secretState = 0
+
+  override def onEvent: Receive = {
+    case p: Pong =>
+      secretState = p.i
+
+    case Ping(i) =>
+      persistOnEvent(Pong(i))
+  }
+
+  override def onCommand: Receive = {
+    case Ping(i) =>
+      persist(Pong(i)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+
+    case Request(request) =>
+      sender ! Response(request)
+
+    case RequestWithPersistence(request, false) =>
+      sender ! Response(request)
+      persist(Response(request)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+
+    case RequestWithPersistence(request, true) =>
+      persist(Response(request)) {
+        case Success(e)  =>
+        case Failure(ex) =>
+      }
+      sender ! Response(request)
+
+    case PersistSlowly(remaining) =>
+      if (remaining.nonEmpty) {
+        self ! PersistSlowly(remaining.tail)
+        persist(Pong(remaining.head)) {
+          case Success(e)  =>
+
+          case Failure(ex) =>
+        }
+
+      }
+  }
+}
+
+class MySimpleActorTestActor(log: ActorRef) extends SimpleActor(log) with ExplicitOnEventCall {
+  override def onCommand: Receive = super.onCommand orElse persistRecieve
+}
+
+/**
+ * fixme: move to reasonable package.
+ */
+class EventsourcedActorTestKitSpec extends TestKit(EventSourcedActorTestKit.actorSystem)
+  with EventSourcedActorTestKit
+  with GivenWhenThen {
+
+  feature("behaviour onCommand is testable") {
+    scenario("test persisted events after call") {
+      Given("an actor")
+      val actor = initEventSourcedActor(new SimpleActor(eventlog))
+
+      Then("persisted events can be tested")
+      // tell actor something
+      actor ! Ping(2)
+      expectEvent(Pong(2))
+
+      // expect event processes communication between event log and actor correctly and is ready for next call.
+      actor ! Ping(3)
+      expectEvent(Pong(3))
+
+    }
+
+    scenario("events are stashed and can be expected after multiple commands") {
+
+      val actor = initEventSourcedActor(new SimpleActor(eventlog))
+
+      actor ! Ping(2)
+      actor ! Ping(3)
+      expectEvent(Pong(2))
+      expectEvent(Pong(3))
+    }
+
+    scenario("test ask pattern") {
+      // implicit execution context and timeout are provided by ActorTest
+      val actor = initEventSourcedActor(new SimpleActor(eventlog))
+
+      val response = actor ? Request("test request")
+      response.map {
+        case Response(s) =>
+          assert(s === "test request")
```

Comment:
scalatest wouldn't receive the error fired by assert as it is thrown asynchronously. Use e.g. scalatest's [Futures](http://doc.scalatest.org/2.2.6/index.html#org.scalatest.concurrent.Futures) or scala's Await to block.